### PR TITLE
refactor: improve memory stats perfomance

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Replace `StatsAlloc` with `tikv-jemalloc-ctl` to get memory stats.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#5613

### CODE REVIEW

File: src/cmd/src/bin/compute_node.rs [hunk 1]
```diff
@@ -14,12 +14,10 @@
 
 #![cfg_attr(coverage, feature(no_coverage))]
 
-use global_stats_alloc::INSTRUMENTED_JEMALLOC;
```
Comment (BugenZhao) [inline src/cmd/src/bin/compute_node.rs:17]: Is this crate still used?

**Review by TennyZhuang — APPROVED** [review]

File: src/stream/src/cache/lru_manager.rs [hunk 2]
```diff
@@ -91,18 +92,27 @@ impl LruManager {
...
 
-            let stats = INSTRUMENTED_JEMALLOC.stats();
-            let cur_total_bytes_used = stats
-                .bytes_allocated
-                .saturating_sub(stats.bytes_deallocated);
+            if let Err(e) = jemalloc_epoch_mib.advance() {
+                tracing::warn!("Jemalloc epoch advance failed! {:?}", e);
+            }
+
+            cur_total_bytes_used = jemalloc_allocated_mib.read().unwrap_or_else(|e| {
```
Comment (BugenZhao) [inline src/stream/src/cache/lru_manager.rs:112]: Have you tested whether the value is correct? Just to make sure it works.

**Review by BugenZhao — APPROVED** [review]
cc @lmatz @chenzl25 Could you please help to do the perfomance tests again after this PR?

File: src/cmd/src/bin/compute_node.rs (hunk 1 above)
Comment (yuhao-su) [reply src/cmd/src/bin/compute_node.rs:17]: No, I will remove it later along with other deprecated crate

File: src/stream/src/cache/lru_manager.rs (hunk 2 above)
Comment (yuhao-su) [reply src/stream/src/cache/lru_manager.rs:112]: Yes.

**Comment by chenzl25** [comment]
> cc @lmatz @chenzl25 Could you please help to do the perfomance tests again after this PR?

Of cause, I already test it on my own laptop and the performance has come back.

**Pull request merged**